plugins/flow-visibility/clickhouse-monitor/main.go

```diff
@@ -0,0 +1,256 @@
+	logString, err := getPodLogs()
+	if err != nil {
+		klog.ErrorS(err, "Not find last monitor job")
+		return false
+	}
+	// A sample log string looks like the following
+	// [clickhouse]host(s)=clickhouse-clickhouse.flow-visibility.svc.cluster.local:9000, database=default, username=clickhouse_operator
+	// ...
+	// [clickhouse][connect=1][prepare] SELECT free_space, total_space FROM system.disks
+	// [clickhouse][connect=1][send query] SELECT free_space, total_space FROM system.disks
+	// ...
+	// I0208 19:54:07.346630       1 main.go:213] "Memory usage" total=1979224064 used=11431936 percentage=0.005775968576744225
+	// I0207 22:29:06.283450       1 main.go:71] "Next round will not be skipped" skipRoundsNum=0
+	// ...
+
+	// reads the number of rounds requires to be skipped
+	logs := strings.Split(logString, "skipRoundsNum=")
+	if len(logs) != 2 {
+		klog.ErrorS(nil, "Error when finding number of rounds")
+		return false
+	}
+	lines := strings.Split(logs[1], "\n")
+	remainingRoundsNum, convErr := strconv.Atoi(lines[0])
+	if convErr != nil {
+		klog.ErrorS(convErr, "Error when finding last monitor job")
+		return false
+	}
+	if remainingRoundsNum > 0 {
+		klog.InfoS("Skip rounds after a successful deletion", "skipRoundsNum", remainingRoundsNum-1)
+		return true
+	}
+	return false
+}
+
+// Gets pod logs from the Clickhouse monitor job
+func getPodLogs() (string, error) {
+	var logString string
+	podLogOpts := corev1.PodLogOptions{}
+	config, err := rest.InClusterConfig()
+	listOptions := metav1.ListOptions{
+		LabelSelector: monitorLabel,
+	}
+	if err != nil {
+		return logString, fmt.Errorf("error when getting config: %v", err)
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return logString, fmt.Errorf("error when getting access to K8S: %v", err)
+	}
+	// gets Clickhouse monitor pod
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), listOptions)
+	if err != nil {
+		return logString, fmt.Errorf("failed to list clickhouse monitor Pods: %v", err)
+	}
+	for _, pod := range pods.Items {
+		// reads logs from the last successful pod
+		if pod.Status.Phase == corev1.PodSucceeded {
+			req := clientset.CoreV1().Pods(namespace).GetLogs(pod.Name, &podLogOpts)
+			podLogs, err := req.Stream(context.TODO())
+			if err != nil {
+				return logString, fmt.Errorf("error when opening stream: %v", err)
+			}
+			defer podLogs.Close()
+
+			buf := new(bytes.Buffer)
+			_, err = io.Copy(buf, podLogs)
+			if err != nil {
+				return logString, fmt.Errorf("error when copying information from podLogs to buf: %v", err)
+			}
+			logString := buf.String()
+			return logString, nil
+		}
+	}
+	return logString, fmt.Errorf("no successful monitor")
+}
+
+// Connects to Clickhouse in a loop
+func connectLoop() (*sql.DB, error) {
+	// Clickhouse configuration
+	userName := os.Getenv("CH_USERNAME")
+	password := os.Getenv("CH_PASSWORD")
+	host, port := os.Getenv("SVC_HOST"), os.Getenv("SVC_PORT")
```

Comment:
nit: In the clickhouse client PR (#3196 ) the DB URI is being passed entirely whereas in this PR is being built from host and port. Not sure if we want to consider a common approach. (this is not something we have to necessarily address in this PR, it's just a suggestion which can be ignored or taken for future work)

Reply:
Thanks Salvatore for reviews! It makes sense for me to use DB URI. Updated the env variables in the monitor.